src/Insurance.sol

```diff
@@ -37,38 +47,45 @@ contract Insurance {
 
     event InsurancePayoutSettled(bytes32 indexed policyId, bytes32 indexed assertionId);
 
+    event InsuredEventSet(bytes32 indexed insuredEvent, uint256 price);
+
     constructor(address _defaultCurrency, address _optimisticOracleV3) {
         defaultCurrency = IERC20(_defaultCurrency);
         oo = OptimisticOracleV3Interface(_optimisticOracleV3);
         defaultIdentifier = oo.defaultIdentifier();
     }
 
-    function issueInsurance(
+    function createPolicy(
         uint256 insuranceAmount,
         address payoutAddress,
         bytes memory insuredEvent
     ) public returns (bytes32 policyId) {
+        require(insuredEventPrice[insuredEvent] > 0, "insuredEvent invalid");
+        require(insuranceAmount <= defaultCurrency.balanceOf(address(this)) - totalLiabilities, "insuranceAmount not available");
         policyId = keccak256(abi.encode(insuredEvent, payoutAddress));
         require(policies[policyId].payoutAddress == address(0), "Policy already exists");
         policies[policyId] = Policy({
             insuranceAmount: insuranceAmount,
             payoutAddress: payoutAddress,
             insuredEvent: insuredEvent,
-            settled: false
+            status: Status.OPEN,
+            assertionId : ''
         });
-        defaultCurrency.safeTransferFrom(msg.sender, address(this), insuranceAmount);
+        defaultCurrency.safeTransferFrom(msg.sender, address(this), insuranceAmount * insuredEventPrice[insuredEvent] / 10000 );
```

Comment:
Is insuredEventPrice[insuredEvent] / 10000 < 1? If so, the settlement would revert as it requires full insuranceAmount payout